### ISSUE TITLE
Make a proposal and try to get as a engine module

### ISSUE THREAD

**Issue by fire** [HungryProton/geometry_helper#1]
This would be great as a engine enhancement, but it needs to have a proposal and converted to c++.

Thoughts?

**Comment by HungryProton** [comment]
I agree, at some point this should be converted to c++. I don't know how to do that yet so for now it's easier for me to add new features as a plugin without having to recompile the entire engine. CSG Polygon edition is not the only thing I'd like to add to this add-on.

I fear the main issue this proposal would face is the port of the PolygonalPath plugin. It's like a regular path node, but constrained to a plane and it holds a polygon representation of the path and a PolygonFinder object for various intents. So either the Path node has to be modified to include these features, but it likely won't happen, or we need to introduce a new node type.
